tests/framework/exec_util.go

```diff
@@ -0,0 +1,143 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package framework
+
+import (
+	"bytes"
+	"io"
+	"net/url"
+	"strings"
+
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/scheme"
+	restclient "k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/remotecommand"
+
+	. "github.com/onsi/gomega"
+)
+
+// ExecOptions passed to ExecWithOptions
+type ExecOptions struct {
+	Command []string
+
+	Namespace     string
+	PodName       string
+	ContainerName string
+
+	Stdin         io.Reader
+	CaptureStdout bool
+	CaptureStderr bool
+	// If false, whitespace in std{err,out} will be removed.
+	PreserveWhitespace bool
+}
+
+// ExecWithOptions executes a command in the specified container,
+// returning stdout, stderr and error. `options` allowed for
+// additional parameters to be passed.
+func (f *Framework) ExecWithOptions(options ExecOptions) (string, string, error) {
+	config, err := LoadConfig()
+	Expect(err).NotTo(HaveOccurred(), "failed to load restclient config")
```

Comment:
you have to decide if framework funcs should call `Expect`. If yes then typically the func name should end with `...OrDie` so it's more obvious that the func may not return. This is especially true of the func calls `Expect` or `Fali` or `panic`  **and** also has an error return value.

Reply:
This would not be relevant either if we add the config to the Framework struct.

Reply:
Can you explain more? I don't understand your reply.

Reply:
Are you thinking of defining a Config struct and embedding it in Framework? If so, consider what the xz compression pkg did where he has a public Config struct, a default constructor, and also a constructor _method_ on Config, where Config fields (non-default) can be set to alter the values set by the constructor. No config args passed, no config struct passed -- pretty clean imo. See [xz](https://github.com/ulikunitz/xz/blob/master/lzma/writer.go#L158)